### PR TITLE
Interface redesign

### PR DESCRIPTION
**Pull request by jamestaylr**
This pull request redesigns the Mapcrafter output interface and address bugs.

  - [x] Adds new styles for visual appeal
  - [x] Removes unused Y coordinate
  - [x] Fixes bug pertaining to `id` (`control-wrapper-x`) assignment by introducing `name()` method to subclasses
  - [x] Remove unused CSS classes

I'm hoping this PR can prompt a discussion. I think Mapcrafter is one of the best (if not the best) Minecraft mappers, but it could use a visual redesign. Components like the compass direction (which this PR preserves), I think, could also be removed.

One downside of the PR is that it introduces an external dependency. These styles, however, could be manually pulled out of the larger library to ensure things properly run offline, while keeping the size of the dependencies down.

Thanks so much for considering the PR!

Image of new interface

### CODE REVIEW

**Comment by m0r13** [comment]
Thanks a lot for your PR! You are right, the Mapcrafter interface could indeed use a visual redesign. These bootstrap buttons look a lot better and I would be glad to merge your PR.

It would be great if you could find a way to store the dependency / needed parts of the dependencies offline, because I really had it as a feature in mind that you can view maps being offline.

Also I wouldn't remove the Y-coordinate, for the isometric view it's good I think to know where X and Z are meant to be on the map (when you see the sea level Y=64 as a reference) and I don't think that it's very confusing for users who don't get that. (As a little advanced option, today I added an option to change this reference Y-coordinate). Maybe it will be even possible to get the actual Y-coordinate belonging to lat/lng coordinate one day.

Two other little things before I merge your PR: Could you please rename the ``.name()`` method to ``getName()`` and fix the merge conflicts (I think I updated the master branch after you forked it) ?

If you have other questions, suggestions, and/or would like to help a bit more on Mapcrafter, just ask, I'm glad to help. :)

**Comment by fenhl** [comment]
Y coordinate needs to stay, even if it's not completely functional in isometric view. Also, the current map type should be visible on the drop-down button. Other than that, bootstrap is a good idea.

**Comment by m0r13** [comment]
Also what do you mean by saying that the compass could be removed?

**Comment by jamestaylr** [comment]
Thanks for the replies!

I completely agree with all the feedback, and I will make the changes over the coming days. Disregard the comment about the compass; it just occurred to me that one map could have multiple rotations. Maybe it would be possible to remove the icon if only one rotation exists? I don't know; I tend to favor minimalism.

I would love to continue helping on this project! If there's anything else the project needs, let me know!

  - [ ] Add Y coordinate
  - [ ] Extract Bootstrap styles for offline use
  - [ ] Change map select label to map name
  - [ ] Rename `name()` to `getName()`
  - [ ] Merge `master` into feature branch to fix merge conflicts

File: src/data/template/static/js/control/html.js [hunk 1]
```diff
@@ -1,12 +0,0 @@
-HTMLControl.prototype = new BaseControl("HTMLControl");
```
Comment (jamestaylr) [inline src/data/template/static/js/control/html.js]: I think it should be fine to remove `HTMLControl`. It's not referenced anywhere else. If you think it should stay, let me know.

Comment (m0r13) [inline src/data/template/static/js/control/html.js]: Should be fine. I can't even remember what I used to use it for.

File: src/data/template/static/js/control/mapselect.js [hunk 2]
```diff
@@ -23,7 +26,10 @@ MapSelectControl.prototype.create = function(wrapper) {
 
 		link.addEventListener("click", (function(ui) {
 			return function(a) {
-				ui.setMap(a.target.getAttribute('data-world'));
+				var h = a.target.getAttribute('data-world');
+				var button = document.getElementById('control-wrapper-map-select').getElementsByTagName('button')[0];
+				button.innerHTML = this.ui.getMapConfig(h).name + ' <span class="caret"></span>';
```
Comment (m0r13) [inline src/data/template/static/js/control/mapselect.js]: I guess ``this.ui`` should be ``ui``

Comment (jamestaylr) [inline src/data/template/static/js/control/mapselect.js]: Ah yeah, you're probably right. It seemed to work in basic testing but `this` should be scoped to the `link` element.

File: src/data/template/static/js/control/mousepos.js [hunk 3]
```diff
@@ -15,6 +15,7 @@ MousePosControl.prototype.create = function(wrapper) {
 			var xzy = ui.latLngToMC(event.latlng, 64);
 			document.getElementById("mouse-move-div").innerHTML = '<div class="btn-group" role="group">'
 			+ '<button type="button" class="btn btn-default">' + "X: " + Math.round(xzy[0]) + '</button>'
+			+ '<button type="button" class="btn btn-default">' + "Y: " + Math.round(xzy[2]) + '</button>'
```
Comment (m0r13) [inline src/data/template/static/js/control/mousepos.js]: Could you please put the buttons in order XZY again? I think that's better as Y is the vertical axis here.

Comment (fenhl) [inline src/data/template/static/js/control/mousepos.js]: I think this is better, since it's how coordinates are displayed everywhere else.

Comment (m0r13) [inline src/data/template/static/js/control/mousepos.js]: Alright then, let's not confuse the users much more.

Comment (jamestaylr) [inline src/data/template/static/js/control/mousepos.js]: I don't really agree with this. I haven't seen it in the reversed order anywhere else, but I have made the changes nonetheless.

Sorry I'm working on this so slowly. I have a lot going on right now.

**Comment by m0r13** [comment]
I see it's possible to download a customized version of bootstrap just with selected plugins and modules: http://getbootstrap.com/customize/

Though I'm not sure if it's really necessary to be super picky about some bootstrap modules, I mean how big is bootstrap, some 100 kbytes? I thought maybe that's not an issue to deliver with the template files.

Any opinions on this? @thorerik @fenhl @jamestaylr

**Comment by jamestaylr** [comment]
Yeah, I think that's a possible quick and easy solution.

I might, however, try to pull out only the styles used in the project. It makes more sense to do it this way, probably, because there's less of a dependence on stuff not really needed.

**Comment by m0r13** [comment]
Are you going to finish the PR or should I take over?

**Comment by jamestaylr** [comment]
I'm out of town for the week, but I will try to rebase it to squash the merge conflicts when I get back.

Sorry for being so late about this.

Then, you should be able to complete the merge and take over.

**Comment by m0r13** [comment]
That's fine, I am mostly busy with other things too unfortunately. Just wanted to know if you are keeping track of it.

**Comment by jamestaylr** [comment]
Okay I tried to look into this tonight and I can't figure out how to get git out of this weird state. I tried to add `mapcrafter/mapcrafter` as the `upstream` `remote`, do rebases, and merge what I thought was the correct `master` into the feature branch.

If you have any suggestions of things I can try, please let me know. I don't have write access to `mapcrafter/mapcrafter`, so that might be the source of the problem. If you want/need write access to my fork, I will be happy to provide that as well.

Sometimes, I literally cannot stand git.

**Comment by m0r13** [comment]
Does this maybe help you? https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request
Not having write-access to this repo shouldn't be a problem, as you are only changing your branch.

**Comment by jamestaylr** [comment]
Unfortunately it does not help. I was actually checking out that article yesterday. 😫

**Comment by m0r13** [comment]
I'm having some issues with rebasing too after synchronizing your branch. I'll just merge your branch as it is.

**Pull request merged**